stdlib/public/Platform/Windows/SwiftWinDbg/SwiftWinDbg.cpp

```diff
@@ -0,0 +1,295 @@
+#define WIN32_LEAN_AND_MEAN // Exclude rarely-used stuff from Windows headers
+#include <windows.h>
+#define KDEXT_64BIT
+#include <wdbgexts.h>
+#include <dbgeng.h>
+
+WINDBG_EXTENSION_APIS ExtensionApis = { 0 };
+EXT_API_VERSION g_ExtApiVersion = { 1, 0, EXT_API_VERSION_NUMBER64, 0 };
+USHORT SavedMajorVersion, SavedMinorVersion;
+
+BOOL APIENTRY DllMain(HMODULE hModule, DWORD ul_reason_for_call,
+                      LPVOID lpReserved) {
+	switch (ul_reason_for_call) {
+	case DLL_PROCESS_ATTACH:
+	case DLL_THREAD_ATTACH:
+	case DLL_THREAD_DETACH:
+	case DLL_PROCESS_DETACH:
+		break;
+	}
+	return TRUE;
+}
+
+LPEXT_API_VERSION WDBGAPI ExtensionApiVersion(void) {
+	return &g_ExtApiVersion;
+}
+
+VOID WDBGAPI WinDbgExtensionDllInit(PWINDBG_EXTENSION_APIS64 lpExtensionApis,
+                                    USHORT MajorVersion, USHORT MinorVersion) {
+  ExtensionApis = *lpExtensionApis;
+  SavedMajorVersion = MajorVersion;
+  SavedMinorVersion = MinorVersion;
+}
+
+DECLARE_API(help) {
+	dprintf("Swift WinDbg Extension\n");
+	dprintf("Author: Ellis Hoag\n\n");
+	// dprintf(" readarray <variable>  - Read the contents of a Swift Array\n");
+  dprintf(" readstring <variable> - Read the contents of a Swift String\n");
+	// dprintf(" readdict <variable>   - Read the contents of a Swift Dictionary\n");
+}
+
+// NOTE: readarray works, but you are better off using Visual Studio to look
+// at arrays.
+//DECLARE_API(readarray) {
```

Comment:
All this code is commented out?

Reply:
It turns out that only the `readstring` function is useful as a WinDbg extension. My plan is to remove that code before I'm finished, but I'm leaving it here as a reference for now.